src/core_plugins/interpreter/public/functions/kibana.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export default () => ({
+  name: 'kibana',
+  type: 'kibana_context',
+  context: {},
+  help: 'Gets kibana global context.',
+  args: {},
+  fn(context) {
+    return {
+      type: 'kibana_context',
+      query: context.query,
```

Comment:
I am confused about that function. How does it get the global Kibana search context in as a context? I thought we inject it as a handler and it needs to read it from that? If it gets it inserted as a context, how could we use this function for adding the global context to a nested expression? E.g.:

```
kibana | esaggs | metric config='{}' fontSize={kibana | escount}
```

Reply:
in executer we pass in the initial context to the pipeline (global filters, query, timerange)
first function in the expression would get that passed in (or first function in subexpression)